app/services/clearancing_service.rb

```diff
@@ -0,0 +1,47 @@
+require 'csv'
+require 'ostruct'
+class ClearancingService
+
+  attr_accessor :clearancing_status
+
+  def initialize
+    self.clearancing_status = create_clearancing_status
+  end
+
+  def process_items(potential_item_ids)
+    potential_item_ids.each do |potential_item_id|
+      item_id_is_valid, error_message = clearancing_status.clearance_batch.item_id_is_valid?(potential_item_id.to_i)
+      if item_id_is_valid
+        clearancing_status.item_ids_to_clearance << potential_item_id.to_i
+      else
+        clearancing_status.errors << error_message
+      end
+    end
+
+    clearance_items!(clearancing_status)
+  end
+
```

Comment:
Do we expect callers to use any method apart from `process_items!`? If not, it might be best to make the rest private to make subsequent refactoring safer.